plugin/trino-hive/pom.xml

```diff
@@ -264,6 +264,12 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
```

Comment:
We recently completed a huge project to [decouple Trino from Hive and Hadoop](https://github.com/trinodb/trino/issues/15921), so we can't reintroduce Hadoop here. Unfortunately, since the Parquet encryption APIs are tied to Hadoop, we'll need to reimplement the decryption code ourselves, inside of Trino, similar to what we have done for most of Parquet and the other file format readers.

Reply:
Hey @electrum I have migrated necessary classes and methods to Trino project and remove the dep on hadoop. I tested with HUDI catalog and it works with encrypted datasets. I could add more unit tests later if the PR looks OK to you. Could you take another look? Thanks!

Reply:
@electrum Bump. No rush tho. Thanks!

Reply:
I see that the Hadoop dependency is still here. We need to remove this.